models/zeroex/zeroex_api_fills_deduped.sql

```diff
@@ -0,0 +1,53 @@
+{{ config(
+        alias ='api_fills_deduped',
+        post_hook='{{ expose_spells(\'["ethereum","arbitrum", "optimism", "polygon","fantom","avalanche_c"]\',
+                                "project",
+                                "zeroex",
+                                \'["rantumBits","bakabhai993"]\') }}'
+        )
+}}
+
+{% set zeroex_models = [  
+ref('zeroex_arbitrum_api_fills_deduped')
+,ref('zeroex_avalanche_c_api_fills_deduped')
+,ref('zeroex_ethereum_api_fills_deduped')
+,ref('zeroex_fantom_api_fills_deduped')
+,ref('zeroex_optimism_api_fills_deduped')
+,ref('zeroex_polygon_api_fills_deduped')
+] %}
+
+
+SELECT *
+FROM (
+    {% for model in zeroex_models %}
+    SELECT
+    volume_usd  as amount_usd,
+      block_date  as block_date,
+      block_time  as block_time,
+      blockchain  as blockchain,
+      evt_index  as evt_index, 
+      maker  as maker, 
+      '0x API'  as project,
+      --contract_address  as project_contract_address,
+      taker  as taker, 
+      maker_symbol as  token_bought_symbol, 
+      token_pair  as token_pair,
+      taker_token as token_sold_address,
+      taker_token_amount  as token_sold_amount,
+      taker_symbol  as  token_sold_symbol,
+      null  as trace_address,
```

Comment:
can you force a datatype when hardcoding `null` on columns? otherwise, spark will infer datatype and applies a `void` type which causes issues. i see two instances in this spell here. example for `trace_address` [here](https://github.com/duneanalytics/spellbook/blob/main/models/oneinch/ethereum/oneinch_v3_ethereum_trades.sql#L44). the `version` column can likely just be string